azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java

```diff
@@ -0,0 +1,124 @@
+import com.microsoft.azure.toolkit.lib.AzureService;
+import com.microsoft.azure.toolkit.lib.SubscriptionScoped;
+import com.microsoft.azure.toolkit.lib.common.entity.CheckNameAvailabilityResultEntity;
+import com.microsoft.azure.toolkit.lib.common.event.AzureOperationEvent;
+import com.microsoft.azure.toolkit.lib.common.model.Subscription;
+import com.microsoft.azure.toolkit.lib.common.operation.AzureOperation;
+import com.microsoft.azure.toolkit.lib.common.task.ICommittable;
+import com.microsoft.azure.toolkit.lib.storage.StorageManagerFactory;
+import com.microsoft.azure.toolkit.lib.storage.model.Kind;
+import com.microsoft.azure.toolkit.lib.storage.model.Performance;
+import com.microsoft.azure.toolkit.lib.storage.model.Redundancy;
+import com.microsoft.azure.toolkit.lib.storage.model.StorageAccountConfig;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class AzureStorageAccount extends SubscriptionScoped<AzureStorageAccount> implements AzureService {
+
+    public AzureStorageAccount() {
+        super(AzureStorageAccount::new);
+    }
+
+    private AzureStorageAccount(@Nonnull final List<Subscription> subscriptions) {
+        super(AzureStorageAccount::new, subscriptions);
+    }
+
+    public List<StorageAccount> list() {
+        return getSubscriptions().stream()
+                .map(subscription -> StorageManagerFactory.create(subscription.getId()))
+                .flatMap(manager -> manager.storageAccounts().list().stream())
+                .map(account -> new StorageAccount(account.manager(), account))
+                .collect(Collectors.toList());
+    }
+
+    public StorageAccount get(@Nonnull String id) {
+        final com.azure.resourcemanager.storage.models.StorageAccount account =
+                StorageManagerFactory.create(ResourceId.fromString(id).subscriptionId()).storageAccounts().getById(id);
+        return new StorageAccount(account.manager(), account);
+    }
+
+    public StorageAccount get(@Nonnull final String resourceGroup, @Nonnull final String name) {
+        final com.azure.resourcemanager.storage.models.StorageAccount account =
+                StorageManagerFactory.create(getDefaultSubscription().getId()).storageAccounts().getByResourceGroup(resourceGroup, name);
+        return new StorageAccount(account.manager(), account);
+    }
+
+    public CheckNameAvailabilityResultEntity checkNameAvailability(String subscriptionId, String name) {
+        StorageManager manager = StorageManagerFactory.create(subscriptionId);
+        CheckNameAvailabilityResult result = manager.storageAccounts().checkNameAvailability(name);
+        return new CheckNameAvailabilityResultEntity(result.isAvailable(),
+                Optional.ofNullable(result.reason()).map(Reason::toString).orElse(null), result.message());
```

Comment:
Optional.ofNullable(result.reason()).map(Reason::toString).orElse(null) -> Objects.toString(result.reason())

Reply:
result may be null.